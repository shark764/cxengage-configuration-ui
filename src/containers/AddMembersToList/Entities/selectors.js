/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import { getModalTableItems } from '../../../redux/modules/entities/listItemSelectors';
import { getModalTableUserItems } from '../../../redux/modules/entities/users/selectors';

export const selectModalTableItems = (state, entityName) => {
  switch (entityName) {
    case 'users': {
      return getModalTableUserItems(state, entityName);
    }
    default: {
      return getModalTableItems(state, entityName);
    }
  }
};
