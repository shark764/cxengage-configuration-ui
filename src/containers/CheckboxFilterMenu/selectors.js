import { createSelector } from 'reselect';

const subState = state => state.get('CheckboxFilterMenu');

const menuItems = (state, props) => {
  if (props.menuType) {
    return state.get('CheckboxFilterMenu').get(props.menuType);
  } else {
    return state.get('CheckboxFilterMenu').get('Columns');
  }
};

export const areAllActive = createSelector(
  [subState, menuItems],
  (subState, menuItems) => {
    let active = true;
    menuItems.forEach(item => (!item.get('active') ? (active = false) : null));
    return active;
  }
);

export const selectColumns = createSelector(subState, subState =>
  subState.get('Columns')
);

export const areAllColNotActive = createSelector(subState, subState => {
  let active = false;
  subState
    .get('Columns')
    .forEach(col => (col.get('active') ? (active = true) : null));
  return active;
});

export const selectTimeFormat = createSelector(subState, subState =>
  subState.get('twelveHourFormat')
);
