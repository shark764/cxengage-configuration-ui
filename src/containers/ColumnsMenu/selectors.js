import { createSelector } from 'reselect';

const selectColumnsMenuMap = state => state.get('ColumnsMenu');

export const selectColumnsMenu = createSelector(
  selectColumnsMenuMap,
  ColumnsMenu => ColumnsMenu
);

export const selectColumnsMenuColumns = createSelector(
  selectColumnsMenuMap,
  subState => {
    return subState.get('columns');
  }
);

export const areAllActive = createSelector(selectColumnsMenuMap, subState => {
  let active = true;
  subState
    .get('columns')
    .forEach(col => (!col.get('active') ? (active = false) : null));
  return active;
});

export const areAllNotActive = createSelector(
  selectColumnsMenuMap,
  subState => {
    let active = false;
    subState
      .get('columns')
      .forEach(col => (col.get('active') ? (active = true) : null));
    return active;
  }
);

export const selectTimeFormat = createSelector(selectColumnsMenuMap, subState =>
  subState.get('twelveHourFormat')
);
