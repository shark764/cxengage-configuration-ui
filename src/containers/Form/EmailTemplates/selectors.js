/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import { Map } from 'immutable';
import {
  getSelectedEntity
} from '../../../redux/modules/crudEndpoint/selectors';

export const getInitialValues = state => {
  const selectedEntity = getSelectedEntity(state);
  const template = selectedEntity.get('template');
  const inherited = selectedEntity.get('inherited');
  let initialValues;
  if (
    template.get('tenantId') ===
    inherited.get('tenantId')
  ) {
    initialValues = {
      email: 'default',
      shared: false,
      subject: inherited.get('subject'),
      body: inherited.get('body')
    };
  } else {
    initialValues = {
      email: 'custom',
      shared: template.get('shared'),
      subject: template.get('subject'),
      body: template.get('body')
    };
  }
  return new Map(initialValues);
}

export const getTemplates = state =>
 getSelectedEntity(state)
  .get('variables')
  .map(variable => variable.get('name'))
  .toJS()
