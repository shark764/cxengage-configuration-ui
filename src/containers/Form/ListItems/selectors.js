/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { createSelector } from 'reselect';
import {
  getSelectedEntity,
  getSelectedSubEntity
} from '../../../redux/modules/crudEndpoint';

export const getFieldItems = createSelector(getSelectedEntity, selectedList =>
  selectedList.getIn(['listType', 'fields']).toJS()
);

// Removes the first attribute from the fields; it is not editable
export const getUpdateFieldItems = createSelector(
  getSelectedEntity,
  selectedList =>
    selectedList
      .getIn(['listType', 'fields'])
      .shift()
      .toJS()
);

// Removes the uneditable first attribute from the initial values and the key
export const getInitialValues = createSelector(
  getSelectedSubEntity,
  selectedListItem =>
    selectedListItem
      .delete('key')
      .delete(selectedListItem.keySeq().toArray()[0])
);
