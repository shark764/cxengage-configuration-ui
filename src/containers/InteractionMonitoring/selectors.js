import { createSelector } from 'reselect';

const selectInteractionMonitoringMap = state =>
  state.get('InteractionMonitoring');

export const selectInteractionMonitoring = createSelector(
  selectInteractionMonitoringMap,
  interactionMonitoring => interactionMonitoring
);

export const selectInteractionMonitoringTableData = createSelector(
  selectInteractionMonitoringMap,
  subState => {
    console.log(subState.toJS());
    return subState.toJS();
  }
  // // console.log(subState.get("tableData").size);
  // if (subState.get("data").size === 0) {
  //   //   console.log(subState.get("tableData"));
  //   return [];
  // } else {
  //   //   console.log(subState.get("tableData"));
  //   return [subState.get("data")];
  // }
  //   }
);
