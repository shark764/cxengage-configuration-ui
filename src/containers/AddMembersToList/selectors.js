/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import { availableEntitiesForList } from '../../redux/modules/entities/selectors';
import { entitiesMetaData } from '../../redux/modules/entities/metaData';
import { availableItemsForList, availableEntityMembersForList } from '../../redux/modules/entities/listItemSelectors';
import { selectAvailableEntityMembersForList } from '../../redux/modules/entities/users/selectors';

export const selectSidePanelTableItems = (state, currentEntity) => {
  const { dependentEntity } = entitiesMetaData[currentEntity];

  switch (currentEntity) {
    case 'outboundIdentifierLists': {
      return availableEntitiesForList(state, dependentEntity, currentEntity);
    }
    case 'skills':
    case 'groups': {
      return availableItemsForList(state, dependentEntity, currentEntity);
    }
    case 'roles': {
      return availableEntityMembersForList(state, dependentEntity, currentEntity);
    }
    case 'dataAccessReports': {
      return selectAvailableEntityMembersForList(state, ['ASSIGNED_REPORTS_READ']);
    }
    default:
      break;
  }
};
