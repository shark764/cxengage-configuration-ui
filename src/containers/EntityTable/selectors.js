/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { entitiesMetaData } from '../../redux/modules/entities/metaData';
import { getAllEntities as getAllEntitiesFromStore, getCurrentEntity } from '../../redux/modules/entities/selectors';
import { getProtectedBranding } from '../../redux/modules/entities/branding/selectors';
import { getRoles } from '../../redux/modules/entities/roles/selectors';

export const getAllEntities = state =>
  getAllEntitiesFromStore(state) ? getAllEntitiesFromStore(state).toJS() : undefined;

export const getAllEntitiesTableData = state => {
  if (getCurrentEntity(state) === 'apiKeys') {
    const roles = getRoles(state) && getRoles(state).get('data');
    return getAllEntitiesFromStore(state)
      ? getAllEntitiesFromStore(state)
          .toJS()
          .map(entity => ({
            ...entity,
            roleName: roles.size > 0 && roles.find(role => role.get('id') === entity.roleId).get('name')
          }))
      : undefined;
  }
  return getAllEntities(state);
};

/**
 * Prepends custom domain to the entity's help link if it exists
 */
export const getHelpLink = state => {
  const entity = entitiesMetaData[getCurrentEntity(state)];
  if (!entity) {
    return undefined;
  }
  // Looking for custom domain to add value to
  // URL to redirect user to proper page
  const customDomain =
    getProtectedBranding(state) &&
    getProtectedBranding(state).find(protectedBranding => protectedBranding.get('key') === 'customDomain');
  const helpURL = `https://${customDomain ? `${customDomain.get('value')}-` : ''}docs.cxengage.net`;
  return `${helpURL}${entity.helpLink}`;
};
