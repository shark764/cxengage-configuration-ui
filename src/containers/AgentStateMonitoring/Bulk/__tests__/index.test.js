/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import AgentStateMonitoringBulkActionsForm, { mapStateToProps, createFormName } from '../';
import { getCurrentEntity } from '../../../../redux/modules/entities/selectors';
import {
  selectPresenceReasonLists,
  countBulkSelectedBusyAgents
} from '../../../../redux/modules/reporting/agentStateMonitoring/selectors';

jest.mock('../../../../redux/modules/entities/selectors');
getCurrentEntity.mockImplementation(() => 'agentStateMonitoring');

jest.mock('../../../../redux/modules/reporting/agentStateMonitoring/selectors');
selectPresenceReasonLists.mockImplementation(() => ({ id: '0001', name: 'reason-list1' }));
countBulkSelectedBusyAgents.mockImplementation(() => 1);

describe('AgentStateMonitoringBulkActionsForm Renders', () => {
  it('renders', () => {
    const store = createStore(state => state);
    expect(
      shallow(<AgentStateMonitoringBulkActionsForm store={store}>Child</AgentStateMonitoringBulkActionsForm>)
    ).toMatchSnapshot();
  });
});

describe('Maps state to props only using selectors', () => {
  it('validates object created from mapStateToProps', () => {
    expect(mapStateToProps()).toMatchSnapshot();
  });
});

describe('createFormName', () => {
  it('returns proper values', () => {
    expect(createFormName()).toMatchSnapshot();
  });
});
