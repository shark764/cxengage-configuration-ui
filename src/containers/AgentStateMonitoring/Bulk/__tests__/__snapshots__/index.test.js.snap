// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentStateMonitoringBulkActionsForm Renders renders 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`Maps state to props only using selectors validates object created from mapStateToProps 1`] = `
Object {
  "countBusyAgentsSelected": 1,
  "key": "agentStateMonitoring:bulk",
  "reasonLists": Object {
    "id": "0001",
    "name": "reason-list1",
  },
  "supervisorUpdatePermissions": Object {
    "direction": undefined,
    "state": undefined,
  },
}
`;

exports[`createFormName returns proper values 1`] = `
Object {
  "form": "agentStateMonitoring:bulk",
}
`;
